### ISSUE TITLE
"Wrong hanging indentation" - misreported, not clearing

### ISSUE THREAD

**Issue by shidarin** [landscapeio/landscape-issues#41]
This bug must have started up in the last week, because it wasn't reported on my repo then, and the lines have not changed.

The following code is reported to have the wrong hanging indentation inside of the help text:

```
parser.add_argument(
    "-i",
    "--input",
    help="specify the filetype to convert from. Use when CDLConvert "
         "cannot determine the filetype automatically. Supported input "
         "formats are: "
         "{inputs}".format(inputs=str(INPUT_FORMATS.keys()))
)
```

I can't get this error to appear in Prospector, pep8, pylint or flake8. It immediately triggers if I change the indentation to an incorrect spacing.

[Landscape Repo Error Page](https://landscape.io/github/shidarin/cdl_convert/26/messages/error)
[Github Repo](https://github.com/shidarin/cdl_convert/tree/develop)

Thanks for your continued help.

**Comment by myint** [comment]
This is from `pylint`. @shidarin, you probably don't have the latest `pylint` installed on your machine. This bug seems like it was introduced by [this](https://bitbucket.org/logilab/pylint/commits/50f940cb47daaa8c3b87a0046bbb97cdf7b6c879) commit.

```
$ pylint --report=no foo.py
...
C:  5, 0: Wrong hanging indentation.
         "cannot determine the filetype automatically. Supported input "
    |    ^ (bad-continuation)
C:  6, 0: Wrong hanging indentation.
         "formats are: "
    |    ^ (bad-continuation)
C:  7, 0: Wrong hanging indentation.
         "{inputs}".format(inputs=str(INPUT_FORMATS.keys()))
    |    ^ (bad-continuation)
...
```

I agree with your assessment though. Your formatting is perfectly fine. `pylint` and landscape should not be complaining about this. Also, it seems strange that landscape identifies this stylistic issue as an "error".

**Comment by shidarin** [comment]
@myint, you're totally right, I upgraded my pylint from 1.2.0 to 1.2.1 and am now getting the note.

Pylint seems to want:

```
parser.add_argument(
    "-i",
    "--input",
    help="specify the filetype to convert from. Use when CDLConvert "
    "cannot determine the filetype automatically. Supported input "
    "formats are: "
    "{inputs}".format(inputs=str(INPUT_FORMATS.keys()))
)
```

I've gone ahead and opened an issue for `pylint` [here](https://bitbucket.org/logilab/pylint/issue/232/wrong-hanging-indentation-false-positive)

The full 1.2.1 changelog is:

```
    * Restore the ability to specify the init-hook option via the
      configuration file, which was accidentally broken in 1.2.0.

    * Add a new warning [bad-continuation] for badly indentend continued
      lines.

    * Emit [assignment-from-none] when the function contains bare returns.
      Fixes BitBucket issue #191.

    * Added a new warning for closing over variables that are
      defined in loops. Fixes Bitbucket issue #176.

    * Do not warn about \u escapes in string literals when Unicode literals
      are used for Python 2.*. Fixes BitBucket issue #151.

    * Extend the checking for unbalanced-tuple-unpacking and
      unpacking-non-sequence to instance attribute unpacking as well.

    * Fix explicit checking of python script (1.2 regression, #219)

    * Restore --init-hook, renamed accidentally into --init-hooks in 1.2.0
      (#211)

    * Add 'indexing-exception' warning, which detects that indexing
      an exception occurs in Python 2 (behaviour removed in Python 3).
```

But since this is a pylint and not a landscape issue, I'll go ahead and close this out.

**Comment by carlio** [comment]
Thanks for the report @shidarin . I agree with @myint , this shouldn't be reported by Landscape even if pylint does, and it certainly shouldn't be an error. I'll reopen this issue to at least change it to a 'style' error.

**Comment by carlio** [comment]
@shidarin Oddly I can't reproduce this locally using pylint 1.2.1

I notice you don't have `\` at the end of the lines, perhaps adding it would also remove the issue? i.e.:

```
parser.add_argument(
    "-i",
    "--input",
    help="specify the filetype to convert from. Use when CDLConvert " \
         "cannot determine the filetype automatically. Supported input " \
         "formats are: " \
         "{inputs}".format(inputs=str(INPUT_FORMATS.keys()))
)
```

**Comment by shidarin** [comment]
I know this is silly, since you probably read this earlier in the thread, but pylint was updated 4 days ago, and only the new version flags this.

I'll throw a gist up with different indent errors and passes in the morning.

I didn't try backslash escaping, I was decently sure pylint would accept that- even if pep8 says you shouldn't. Instead I specifically disabled C0330 on all 5 erroring indented lines. Gotta take a stand somewhere.

**Comment by myint** [comment]
@carlio, I agree with @shidarin, adding unnecessary backslashes is bad form.

By the way, I think the `pep8` tool seems far superior to `pylint` for stylistic issues.

**Comment by shidarin** [comment]
Created this [gist](https://gist.github.com/shidarin/06e59191ab0c7f26c1c3) to have a nice example file.

This triggers the following errors for me in pylint 0.2.1:

```
C: 28, 0: Wrong hanging indentation.
             "despite it's pleasing looks and lack of superfluous slashes."
        |    ^ (bad-continuation)
C: 34, 0: Wrong hanging indentation.
            "like this."
        |   ^ (bad-continuation)
C:  1, 0: Missing module docstring (missing-docstring)
C:  4, 0: Missing function docstring (missing-docstring
```

**Comment by myint** [comment]
This has been fixed in pylint's default branch.

https://bitbucket.org/logilab/pylint/commits/e3ecc895733cdf6ea3c3822abcb58cf08feb7184

**Comment by carlio** [comment]
@myint Ah great! I'll update Landscape once that hits PyPI :)

**Comment by myint** [comment]
This particular issue should be fixed in pylint 1.3 now. Though, I found other [related instances](https://bitbucket.org/logilab/pylint/issue/289/false-positive-for-bad-continuation), which I've reported.

**Comment by carlio** [comment]
Landscape has been using pylint 1.3 for a while now.
